api/src/main/java/org/openmrs/module/ugandaemrreports/reports2019/SetupANCRegister2019.java

```diff
@@ -0,0 +1,224 @@
+        dsd.addColumn("VL", sdd.getWHOCD4ViralLoadCalculation("dc8d83e3-30ab-102d-86b0-7a5022ba4115", "0b434cfa-b11c-4d14-aaa2-9aed6ca2da88"), "onDate=${endDate}", new CalculationResultDataConverter());
+        dsd.addColumn("Pre-ART No", sdd.getPreARTNumber(), "");
+        dsd.addColumn("IYCF", sdd.definition("IYCF", sdd.getConcept("5d993591-9334-43d9-a208-11b10adfad85")), "onOrAfter=${startDate},onOrBefore=${endDate}", new IYCFDataConverter());
+        dsd.addColumn("Haemoglobin", sdd.definition("Haemoglobin", sdd.getConcept("dc548e89-30ab-102d-86b0-7a5022ba4115")), "onOrAfter=${startDate},onOrBefore=${endDate}", new ObsDataConverter());
+        dsd.addColumn("Syphilis testW", sdd.definition("Syphilis testW", sdd.getConcept("3c5aa2a6-ca7a-478c-b559-d11040691b8d")), "onOrAfter=${startDate},onOrBefore=${endDate}",new ObsDataConverter());
+        dsd.addColumn("Syphilis testP", sdd.definition("Syphilis testP", sdd.getConcept("3c5aa2a6-ca7a-478c-b559-d11040691b8d")), "onOrAfter=${startDate},onOrBefore=${endDate}", new ObsDataConverter());
+        dsd.addColumn("FPC", sdd.definition("FPC", sdd.getConcept("0815c786-5994-49e4-aa07-28b662b0e428")), "onOrAfter=${startDate},onOrBefore=${endDate}", new FpcDataConverter());
+        dsd.addColumn("TT", sdd.definition("TT", sdd.getConcept("39217e3d-6a39-4679-bf56-f0954a7ffdb8")), "onOrAfter=${startDate},onOrBefore=${endDate}", new TetanusDataConverter());
+        dsd.addColumn("IPT/CTX", sdd.definition("IPT/CTX", sdd.getConcept("1da3cb98-59d8-4bfd-b0bb-c9c1bcd058c6")), "onOrAfter=${startDate},onOrBefore=${endDate}", new IptCtxDataConverter());
+        dsd.addColumn("Free LLIN", sdd.definition("Free LLIN", sdd.getConcept("3e7bb52c-e6ae-4a0b-bce0-3b36286e8658")), "onOrAfter=${startDate},onOrBefore=${endDate}", new FreeLlinDataConverter());
+        dsd.addColumn("Mebendazole", sdd.definition("Mebendazole", sdd.getConcept("9d6abbc4-707a-4ec7-a32a-4090b1c3af87")), "onOrAfter=${startDate},onOrBefore=${endDate}", new MebendazoleDataConverter());
+        dsd.addColumn("Iron given", sdd.getIronGiven(), "onDate=${endDate}", new CalculationResultDataConverter());
+        dsd.addColumn("Folic acid given", sdd.getFolicAcidGiven(), "onDate=${endDate}", new CalculationResultDataConverter());
+        dsd.addColumn("Other treatments", sdd.definition("Other treatments", sdd.getConcept("2aa72406-436e-490d-8aa4-d5336148204f")), "onOrAfter=${startDate},onOrBefore=${endDate}", new ObsDataConverter());
+        dsd.addColumn("Referal In/Out", sdd.referredToOrFrom(), "onDate=${endDate}", new CalculationResultDataConverter());
+        dsd.addColumn("Risk Factor/Complications", sdd.definition("Risk Factor/Complications", sdd.getConcept("120186AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA")), "onOrAfter=${startDate},onOrBefore=${endDate}", new ObsDataConverter());
+
+        return dsd;
+    }
+    private DataDefinition getEncounterDate() {
```

Comment:
@solemabrothers This needs to be from a global definition

Reply:
Done

Reply:
@solemabrothers You do not need a private method for this. 

Also what is the difference between ANCEncounterCalculation and other encounter calculations?